From bcecb2ae3fc30614a207f05d6a4a754fb2335740 Mon Sep 17 00:00:00 2001
From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Thu, 10 Jan 2019 11:12:32 +0100
Subject: [PATCH v1 3/3] drivers/tty: increase priority for tty_buffer_worker

sched_priority = 1 is enough to dramatically reduce latency
on have system load produced by tasks with default user space prio.

Signed-off-by: Oleksij Rempel <o.rempel@pengutronix.de>
Tested-by: Alexander Sverdlin <alexander.sverdlin@gmail.com>
---
 drivers/tty/tty_buffer.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/tty/tty_buffer.c b/drivers/tty/tty_buffer.c
index 3b041aa9b72a..df06e780d9d4 100644
--- a/drivers/tty/tty_buffer.c
+++ b/drivers/tty/tty_buffer.c
@@ -13,11 +13,13 @@
 #include <linux/string.h>
 #include <linux/slab.h>
 #include <linux/sched.h>
+#include <linux/sched/rt.h>
 #include <linux/wait.h>
 #include <linux/bitops.h>
 #include <linux/delay.h>
 #include <linux/module.h>
 #include <linux/ratelimit.h>
+#include <uapi/linux/sched/types.h>
 
 
 #define MIN_TTYB_SIZE	256
@@ -562,12 +564,16 @@ bool tty_buffer_queue_work(struct tty_port *port)
 
 void tty_buffer_init_kthread(void)
 {
+	struct sched_param param = { .sched_priority = 1 };
+
 	tty_buffer_worker = kthread_create_worker(0, "tty");
 	if (IS_ERR(tty_buffer_worker)) {
 		pr_err("Failed to create tty buffer kworker\n");
 		return;
 	}
+
+	sched_setscheduler(tty_buffer_worker->task, SCHED_FIFO, &param);
 }
 
 /**
 
-- 
2.7.4

